File: euler_033.py
from time import time
from math import gcd
from numpy import prod

def check_fraction(frac):
    num = frac[0]
    den = frac[1]
    n = sorted(str(num))
    d = sorted(str(den))
    if n[0] == d[0]:
        return num*int(d[1]) == den*int(n[1])
    if n[1] == d[1]:
        return num * int(d[0]) == den * int(n[0])
    return False


start = time()

fractions = set()

for num in [n for n in range(10,100) if n%10 > 0]:
    for den in range(num + 1,100):
        if check_fraction((num,den)):
            fractions.add((num,den))

num_prod = prod([f[0] for f in fractions])
den_prod = prod([f[1] for f in fractions])

print("The denominator is: " + str(den_prod // gcd(num_prod, den_prod)))

stop = time()
print("Time: " + str(stop-start))

File: euler_033.js
const start = Date.now();

const gcd = (a,b) => {
    if (a < b) {
        return gcd(b,a);
    }
    let r = a % b;
    while (r > 0) {
        a = b;
        b = r;
        r = a % b;
    }
    return b;
}

let allFractions = [];

for (let a = 1; a < 10; a++) {
    for (let b = 0; b < 10; b++) {
        for (let c = 1; c < b; c++) {
            allFractions.push([10 * b + a, 10 * a + c, b, c]);
            allFractions.push([10 * a + b, 10 * c + a, b, c]);
        }
    }
}

let sameValueFractions = [];

for (const f of allFractions) {
    if (f[0] * f[3] === f[1] * f[2]) {
        sameValueFractions.push(f);
    }
}

let numerator = 1;
let denominator = 1;
for (const f of sameValueFractions) {
    numerator *= f[1];
    denominator *= f[0];
}

const res = denominator / gcd(numerator,denominator);

const end = Date.now();

console.log("Execute time: ", end - start, " Denominator is: ", res);